### DISCUSSION TITLE
Azure Isolated Function template: why don't we use the DI container ?

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.templates#818] [General]
In the Isolated Function template, we pre-provision a sample Azure Function.
This function uses a logger, but instead of having that logger injected into the function, we're creating it ourselves.  Why do we do that ?  Why don't we use the DI container for this ?

In other words, we're currently doing this:

```csharp
ILogger logger = context.GetLogger<OrderFunction>();
logger.LogInformation("C# HTTP trigger 'order' function processed a request");
```
([source](https://github.com/arcus-azure/arcus.templates/blob/f05c0c8e64e4c5203c05c5f982b72ddd4d52502b/src/Arcus.Templates.AzureFunctions.Http/OrderFunction.cs#LL147C13-L148C91))

Why don't we just inject the ILogger via the constructor ?

**Comment by stijnmoreels** [comment]
Ah, probably bc this was also done in the regular Microsoft Azure Functions project template.
I'm guessing the difference here is that the logger will be created using the scoped request services instead of the regular ones, but otherwise, I'll have to look at the code.

So, mainly streamlining with Microsoft default here, I guess.

**Reply by fgheysels** [reply]
I think a DI approach is more 'best practice', don't you think ?

**Reply by stijnmoreels** [reply]
Yes, I would think so too, but their templates in both in-process and isolated are retrieving the logger instance in a slightly different way. It is still a logger from the DI container, only retrieved in a scoped fashion.

**Reply by fgheysels** [reply]
Ok.  I meant:  we should inject it in the ctor, instead of retrieving it like we do now :)

**Reply by stijnmoreels** [reply]
😄 , depends on what effect that has on the request logging, maybe. If we still have the same telemetry as before. Otherwise, yeah 👍
